src/main/scala/org/ergoplatform/ErgoApp.scala

```diff
@@ -65,15 +67,30 @@ class ErgoApp(args: Seq[String]) extends Application {
     txGen ! StartGeneration
   }
 
+  val actorsToStop = Seq(minerRef,
+    peerManagerRef,
+    networkControllerRef,
+    readersHolderRef,
+    nodeViewSynchronizer,
+    localInterface,
+    nodeViewHolderRef
+  )
+  sys.addShutdownHook(ErgoApp.shutdown(actorSystem, actorsToStop))
```

Comment:
I think we also need to ensure that shutdown procedure is correct for the node terminations due to unhandled exceptions, e.g. in Actor threads. Coordinated shutdown occurs in this case, but I'm not sure that we close all the actors gracefully.